tests/test_apply.py

```diff
@@ -243,6 +243,33 @@ def test_dry_run(monkeypatch):
         s3.head_bucket(Bucket=bucket_name)
 
 
+def test_service_endpoint_alias_replacements(monkeypatch):
+    monkeypatch.setenv("DRY_RUN", "1")
+    config = """
+    provider "aws" {
+      region = "eu-west-1"
+    }"""
+
+    temp_dir = deploy_tf_script(config, cleanup=False, user_input="yes")
+    override_file = os.path.join(temp_dir, "localstack_providers_override.tf")
+    assert check_override_file_content(override_file)
+    rmtree(temp_dir)
+
+
+def check_override_file_content(override_file):
+    try:
+        with open(override_file, "r") as fp:
+            result = hcl2.load(fp)
+            result = result["provider"][0]["aws"]
+    except Exception as e:
+        print(f'Unable to parse "{override_file}" as HCL file: {e}')
```

Comment:
I think we should raise an exception here right away (otherwise may fail with `NameError` in line 267 below, if `result` has not been assigned value.

```suggestion
        raise Exception(f'Unable to parse "{override_file}" as HCL file: {e}')
```